File: offgame-core/src/main/java/org/retal/offgame/dto/UnitDetails.java
package org.retal.offgame.dto;

import lombok.*;
import org.retal.offgame.entity.units.EngineType;
import org.retal.offgame.entity.units.UnitType;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UnitDetails {

    private Long id;
    private String name;
    private String imageName;
    private String description;
    private Long hull;
    private Long shields;
    private Long attack;
    private Long capacity;
    private EngineType engineType;
    private UnitType unitType;
    private Long speed;
    private Long fuelConsumption;
    private List<RequirementDTO> requirements;
}

File: offgame-ui/src/app/model/UnitDetails.ts
import {Requirement} from "./Requirement";

export class UnitDetails {

    id!: number
    description!: string
    imageName!: string
    name!: string
    hull!: number
    shields!: number
    attack!: number
    capacity!: number
    engineType!: string
    unitType!: string
    speed!: number
    fuelConsumption!: number
    requirements: Requirement[]

    constructor(data: any) {
        Object.assign(this, data);

        this.requirements = data.requirements.map((object: any) => new Requirement(object))
    }

    getEngineType(): string {
        switch (this.engineType) {
            case "reactive":
                return "Реактивный"
            case "impulse":
                return "Импульсный"
            case "hyperspace":
                return "Гиперпространственный"
            default:
                return "только маневровые"
        }
    }
}